File: java/src/main/java/com/adaptionsoft/games/uglytrivia/domain/events/GoldCoinWon.java
package com.adaptionsoft.games.uglytrivia.domain.events;

import java.util.Objects;

public class GoldCoinWon {
    public final String playerName;
    public final int goldCoins;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        GoldCoinWon that = (GoldCoinWon) o;
        return goldCoins == that.goldCoins && Objects.equals(playerName, that.playerName);
    }

    @Override
    public int hashCode() {
        return Objects.hash(playerName, goldCoins);
    }

    public GoldCoinWon(String playerName, int goldCoins) {
        this.playerName = playerName;
        this.goldCoins = goldCoins;
    }
}

File: php/src/Domain/Event/GoldCoinWon.php
<?php


namespace Trivia\Domain\Event;


class GoldCoinWon
{
    public $playerName;
    /**
     * @var int
     */
    public $goldCoins;

    /**
     * GoldCoinWon constructor.
     * @param $playerName
     * @param int $goldCoins
     */
    public function __construct($playerName, int $goldCoins)
    {
        $this->playerName = $playerName;
        $this->goldCoins = $goldCoins;
    }
}